spec.md

```diff
@@ -19,15 +19,246 @@
+
+#### (HTTP API) GET `/spaces/`
+
+Example request:
+
+```http
+GET /spaces/ HTTP/1.1
+Host: example.com
+Accept: application/json
+Authorization: ...
+```
+
+Example success response (requester has read access to at least one space):
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+...
+```
+
+Example success response (requester does NOT have access to any spaces):
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+...
+```
+
+Example error response (missing authorization):
+
+### Update Space operation
+
+* Requires appropriate authorization (root zcap invoked by the space's controller,
+  or a zcap granting permission to write to a particular space)
+* Allows to update the following fields:
+  - `name`
+  - `controller`
+
+#### (HTTP API) PUT `/spaces/{space_id}`
+
+Note that this is a _full_ update (partial updates via http `PATCH` verb might
+be supported later). However, some fields may not be updated (like `id`) and so
+may be omitted from the request payload.
+
+Note that this operation is idempotent.
+
+Example request (updating the name of a space):
+
+```http
+PUT /spaces/81246131-69a4-45ab-9bff-9c946b59cf2e HTTP/1.1
+Host: example.com
+Content-type: application/json
+Accept: application/json
+Authorization: ...
+
+{
+  "id": "81246131-69a4-45ab-9bff-9c946b59cf2e",
+  "name": "Newly renamed space #1",
+  "controller": "did:key:z6MkpBMbMaRSv5nsgifRAwEKvHHoiKDMhiAHShTFNmkJNdVW"
+}
+```
+
+Example success response (returns the updated details of the space):
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+Location: https://example.com/spaces/81246131-69a4-45ab-9bff-9c946b59cf2e
+
+{
+  "id": "81246131-69a4-45ab-9bff-9c946b59cf2e",
+  "type": ["Space"],
+  "name": "Newly renamed space #1",
+  "controller": "did:key:z6MkpBMbMaRSv5nsgifRAwEKvHHoiKDMhiAHShTFNmkJNdVW"
+}
+```
+
+Example error response (missing or invalid authorization):
+
+Example error response (invalid `id` provided, the space does not exist):
+
+Example error response (client is attempting to change an immutable field like
+the space `id`):
+
+### Delete Space operation
+
+* Requires appropriate authorization (root zcap invoked by the space's controller,
+  or a zcap granting permission to write to a particular space)
+* Deletes the space and all of the data (collections and resources) contained
+  in it
+* Server responds with a Tombstone object (which will be later used for replication)
+* This operation is idempotent
+
+#### (HTTP API) DELETE `/spaces/{space_id}`
+
+Example request (no request body):
+
+```http
+DELETE /spaces/81246131-69a4-45ab-9bff-9c946b59cf2e HTTP/1.1
+Host: example.com
+Accept: application/json
+Authorization: ...
+```
+
+Example success response:
+
+```http
+HTTP/1.1 200 OK
+Content-type: application/json
+
+{
+  ... TODO: specify Tombstone format
+}
+```
+
+TODO: Decide whether subsequent GET requests to the same space should result in
+a 410 Gone, or the usual 404.
```

Comment:
imho 404 not 410.

Reply:
ok but then how else am I gonna prove my http hipster status (if I don't suggest 410)? 
oooh i know, I'll mint a self-issued VC giving me hipster status...

Reply:
(I joke, but there's arguments for 410, but we can discuss separately)